src/pip/_internal/collector.py

```diff
@@ -15,12 +15,9 @@
 from pip._vendor.six.moves.urllib import request as urllib_request
 
 from pip._internal.models.link import Link
-from pip._internal.utils.misc import (
-    ARCHIVE_EXTENSIONS,
-    path_to_url,
-    redact_auth_from_url,
-)
+from pip._internal.utils.misc import path_to_url, redact_auth_from_url
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+from pip._internal.utils.unpacking import ARCHIVE_EXTENSIONS
```

Comment:
Actually, it might be better not to put `ARCHIVE_EXTENSIONS` in `unpacking.py` and related constants so that e.g. `collector.py`, `index.py`, `link.py`, etc. don't need to import `unpacking` just to get access to a list of extensions (since they don't actually need to unpack anything).

Reply:
I think `SUPPORTED_EXTENSIONS` is the only one where there's an argument to move it to `unpacking.py`, since it requires doing extra "work" (extra imports) to define it.

Reply:
What about putting `ARCHIVE_EXTENSIONS` and friends in `utils.filetypes` and `SUPPORTED_EXTENSIONS` in `utils.unpacking`?

Reply:
That sounds fine.